chef-config/lib/chef-config/config.rb

```diff
@@ -305,6 +308,30 @@ def self.path_accessible?(path)
     default :diff_output_threshold,   1000000
     default :local_mode, false
 
+    # Configures the mode of operation for ChefFS, which is applied to the
+    # ChefFS-based knife commands and chef-client's local mode. (ChefFS-based
+    # knife commands include: knife delete, knife deps, knife diff, knife down,
+    # knife edit, knife list, knife show, knife upload, and knife xargs.)
+    #
+    # Valid values are:
+    # * "static": ChefFS only manages objects that exist in a traditional Chef
+    #   Repo as of Chef 11.
+    # * "everything": ChefFS manages all object types that existed on the OSS
+    #   Chef 11 server.
+    # * "hosted_everything": ChefFS manages all object types as of the Chef 12
+    #   Server, including RBAC objects and Policyfile objects (new to Chef 12).
+    #
+    # TODO: add a test
+    default :repo_mode do
+      if local_mode
```

Comment:
I would make this `if local_mode && !chef_zero.osc_compat`, so that if someone *does* go into osc_compat mode, they don't get errors.